esphome/components/seeed_mr60fda2/seeed_mr60fda2.cpp

```diff
@@ -0,0 +1,396 @@
+        select_index_ = find_nearest_index(install_height_float, INSTALL_HEIGHT, 7);
+        this->install_height_select_->publish_state(this->install_height_select_->at(select_index_).value());
+      }
+
+      if (this->height_threshold_select_ != nullptr) {
+        this->current_height_threshold_int_ =
+            encode_uint32(current_data_buf_[7], current_data_buf_[6], current_data_buf_[5], current_data_buf_[4]);
+
+        height_threshold_float = bit_cast<float>(current_height_threshold_int_);
+        select_index_ = find_nearest_index(height_threshold_float, HEIGHT_THRESHOLD, 7);
+        this->height_threshold_select_->publish_state(this->height_threshold_select_->at(select_index_).value());
+      }
+
+      if (this->sensitivity_select_ != nullptr) {
+        this->current_sensitivity_ =
+            encode_uint32(current_data_buf_[11], current_data_buf_[10], current_data_buf_[9], current_data_buf_[8]);
+
+        select_index_ = find_nearest_index(this->current_sensitivity_, SENSITIVITY, 3);
+        this->sensitivity_select_->publish_state(this->sensitivity_select_->at(select_index_).value());
+      }
+
+      ESP_LOGD(TAG, "Mounting height: %.2f, Height threshold: %.2f, Sensitivity: %u", install_height_float,
+               height_threshold_float, this->current_sensitivity_);
+      this->current_frame_locate_ = LOCATE_FRAME_HEADER;
+      break;
+
+    default:
+      break;
+  }
+}
+
+// Sending data frames
+void MR60FDA2Component::send_query_(uint8_t *query, size_t string_length) { this->write_array(query, string_length); }
+
+/**
+ * @brief Convert a float value to a byte array.
+ *
+ * This function converts a float value to a byte array.
+ *
+ * @param value The float value to convert.
+ * @param bytes The byte array to store the converted value.
+ */
+void MR60FDA2Component::float_to_bytes_(float value, unsigned char *bytes) {
+  union {
+    float float_value;
+    unsigned char byte_array[4];
+  } u;
+
+  u.float_value = value;
+  memcpy(bytes, u.byte_array, 4);
+}
+
+/**
+ * @brief Convert a 32-bit unsigned integer to a byte array.
+ *
+ * This function converts a 32-bit unsigned integer to a byte array.
+ *
+ * @param value The 32-bit unsigned integer to convert.
+ * @param bytes The byte array to store the converted value.
+ */
+void MR60FDA2Component::int_to_bytes_(uint32_t value, unsigned char *bytes) {
+  bytes[0] = value & 0xFF;
+  bytes[1] = (value >> 8) & 0xFF;
+  bytes[2] = (value >> 16) & 0xFF;
+  bytes[3] = (value >> 24) & 0xFF;
+}
+
+// Send Heartbeat Packet Command
+void MR60FDA2Component::set_install_height(uint8_t index) {
+  uint8_t send_data[13] = {0x01, 0x00, 0x00, 0x00, 0x04, 0x0E, 0x04, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00};
+  uint8_t data_frame[4] = {0x00, 0x00, 0x00, 0x00};
+
+  float_to_bytes_(INSTALL_HEIGHT[index], &send_data[8]);
+
+  for (int i = 0; i < 4; i++) {
+    data_frame[i] = send_data[i + 8];
+  }
+
+  send_data[12] = calculate_checksum(data_frame, 4);
```

Comment:
The entire `data_frame` array is not necessary
```suggestion
  float_to_bytes_(INSTALL_HEIGHT[index], &send_data[8]);
  send_data[12] = calculate_checksum(send_data + 8, 4);
```

Reply:
This same code is in multiple places 
I just dont always put the suggestions as it forces you the contributor to learn and check the rest of the code again.